### PR TITLE
Migrate to node 15 npm 7

### PR DESCRIPTION
**Pull request by MikeVitik**

### CODE REVIEW

**Review by ivanblinov2k17 — DISMISSED** [review]
is package lock commit intended?

File: packages/tests/unit-tests/helpers/webpack-test-compiler.ts [hunk 1]
```diff
@@ -39,10 +39,11 @@ export default (
   return new Promise<webpack.Stats>((resolve, reject) => {
     compiler.run((err, stats) => {
       if (err) reject(err);
-      if (stats.hasErrors())
-        reject(new Error(stats.toJson().errors.join("\n")));
-
-      resolve(stats);
+      if (stats) {
```
Comment (dxvladislavvolkov) [inline packages/tests/unit-tests/helpers/webpack-test-compiler.ts:42]: Can we use something like this?
if(stats && stats.hasErrors())

Comment (MikeVitik) [reply packages/tests/unit-tests/helpers/webpack-test-compiler.ts:42]: Yes. I change to stats?.hasError()

File: packages/runtime/gulpfile.js [hunk 2]
```diff
@@ -13,20 +13,18 @@ platforms.forEach(function(platform) {
   });
 });
 
-gulp.task('compile.esm', function(done) {
-  gulp
-    .src(['./**/*.ts', './**/*.tsx'])
-    .pipe(ts('tsconfig.esm.build.json'))
+gulp.task('compile.esm', function() {
+  var tsProject = ts.createProject('tsconfig.esm.build.json');
```
Comment (dxvladislavvolkov) [inline packages/runtime/gulpfile.js:17]: const tsProject = ts.createProject('tsconfig.esm.build.json');

**Review by dxvladislavvolkov — CHANGES_REQUESTED** [review]

**Review by dxvladislavvolkov — APPROVED** [review]

**Pull request merged**